crates/red_knot_python_semantic/src/types.rs

```diff
@@ -758,17 +747,17 @@ impl<'db> Type<'db> {
             Type::Never => Type::Never,
             Type::Instance(class) => Type::Class(*class),
             Type::Union(union) => union.map(db, |ty| ty.to_meta_type(db)),
-            Type::BooleanLiteral(_) => builtins_symbol_ty(db, "bool"),
-            Type::BytesLiteral(_) => builtins_symbol_ty(db, "bytes"),
-            Type::IntLiteral(_) => builtins_symbol_ty(db, "int"),
+            Type::BooleanLiteral(_) => KnownClass::Bool.to_class(db),
+            Type::BytesLiteral(_) => KnownClass::Bytes.to_class(db),
+            Type::IntLiteral(_) => KnownClass::Int.to_class(db),
             Type::Function(_) => types_symbol_ty(db, "FunctionType"),
             Type::Module(_) => types_symbol_ty(db, "ModuleType"),
-            Type::Tuple(_) => builtins_symbol_ty(db, "tuple"),
+            Type::Tuple(_) => KnownClass::Tuple.to_class(db),
             Type::None => typeshed_symbol_ty(db, "NoneType"),
             // TODO not accurate if there's a custom metaclass...
             Type::Class(_) => builtins_symbol_ty(db, "type"),
```

Comment:
Nit: I'd probably add the other "fundamental" types in this list (`types.ModuleType`, `types.FunctionType`, `_typeshed.NoneType` and `builtins.type`) to the `KnownClass` enum as well, and take `typing.Any` out of the enum for now

Reply:
Why is it `_typeshed.NoneType`? In my repl I see the `type(None)` from builtins?

Reply:
The type of `None` (`NoneType`) is not exposed in the stdlib at all in some versions of Python, and is never exposed in builtins. The `None` singleton object is in builtins, but typeshed needs a place to give the definition of the _type_.

Reply:
Would you have a reference or strategy (maybe to drop in the comment of the enum) to find out where a given common symbol is defined?

Reply:
The [`_typeshed` module](https://github.com/python/typeshed/tree/main/stdlib/_typeshed) is a fictitious module that typeshed pretends is part of the standard library. Because typeshed pretends it's part of the stdlib, and we treat typeshed's stubs as the sole source of truth for the stdlib, we think it's part of the stdlib too!

The purpose of the `_typeshed` module is twofold:
- to define various convenient aliases and protocols that don't exist at runtime but which are used extensively in typeshed's stubs for the stdlib. These need to be defined in a stdlib module, because typeshed's stubs for the stdlib can't depend on any non-stdlib stubs.
- to define certain compatibility workarounds for type checkers (that's us!) to use. `_typeshed.NoneType` is an example of this. At runtime, the type of `None` is `types.NoneType`, but `types.NoneType` was [only added in 3.10](https://docs.python.org/3/library/types.html#types.NoneType). We can't use that as the type of `None`, as we still support Python 3.8 and 3.9, and all other type checkers also have this problem -- so typeshed defines `_typeshed.NoneType` as a little compatibility hack for us.

Reply:
So, small question on the topic. I see that we have methods to lookup `typing_extensions`, but nothing from `typing`

Reply:
> Would you have a reference or strategy (maybe to drop in the comment of the enum) to find out where a given common symbol is defined?

I guess I'd probably `cd` into this directory and grep for the type's name? https://github.com/astral-sh/ruff/tree/main/crates/red_knot_vendored/vendor/typeshed/stdlib

Reply:
> So, small question on the topic. I see that we have methods to lookup `typing_extensions`, but nothing from `typing`

Yeah, that's for a very similar reason! There's some symbols that only exist in `typing` on some versions of Python, but exist in `typing_extensions` on all versions of Python. That means that it's usually better for a type checker to look things up in `typing_extensions` rather than `typing`, since `typing_extensions` re-exports all symbols from `typing`, and typeshed [pretends `typing_extensions` is part of the stdlib](https://github.com/python/typeshed/blob/main/stdlib/typing_extensions.pyi). (I _promise_ typeshed doesn't do this for _loads_ of modules... only `_typeshed` and `typing_extensions`!!)